sstables/mp_row_consumer.hh

```diff
@@ -1087,9 +1079,11 @@ public:
         if (!_is_mutation_end) {
             return proceed::yes;
         }
-        _mutation = new_mutation{partition_key::from_exploded(key.explode(*_schema)), tombstone(deltime)};
-        setup_for_partition(_mutation->key);
-        return proceed::no;
+        auto pk = partition_key::from_exploded(key.explode(*_schema));
+        setup_for_partition(pk);
+        auto dk = dht::global_partitioner().decorate_key(*_schema, pk);
+        _reader->on_next_partition(std::move(dk), tombstone(deltime));
+        return proceed::yes;
     }
```

Comment:
I don't understand why the code is repeated in consume_row_start(). After consume_partition_start(), shouldn't _mutation_end be false?

Reply:
Are you referring to `consume_row_start()` from the KA/LA reader (`mp_row_consumer_k_l`)? This one is for the MC reader (`mp_row_consumer_m`). Both methods handle partition start, hence the duplication, but in different formats.

Reply:
I see.